Add TXT export option for resumes

AnshSinghSonkhia opened the pull request:
Introduces the ability to export resumes as plain text (.txt) files. Updates the UI to include a TXT download button, adds the necessary icon, and implements export logic in ExporterService. Also updates the README to mention TXT export support.

This solves #56 

---

# Screenshot
<img width="1918" height="1027" alt="txt export feature" src="https://github.com/user-attachments/assets/31104e3c-325c-4bc0-8fdf-104ad1000dfe" />

# Sample Exported `.txt` file
[Ansh Singh Sonkhia_Junior Software Engineer.txt](https://github.com/user-attachments/files/21391582/Ansh.Singh.Sonkhia_Junior.Software.Engineer.txt)




    
<!-- This is an auto-generated description by cubic. -->
---

## Summary by cubic
Added the option to export resumes as plain text (.txt) files. The UI now includes a TXT download button, and TXT export is supported in the ExporterService and README.

- **New Features**
  - TXT export button added to the editor menu.
  - TXT export logic implemented to generate plain text resumes.
  - TXT export support mentioned in the README.

<!-- End of auto-generated description by cubic. -->

AnshSinghSonkhia reviewed (commented):
I believe it is good to go now.

gianfrancodemarco commented on src/views/EditorPage.vue:
```diff
@@ -287,6 +289,39 @@ const handleHome = () => {
     // Navigate to home page
     router.push('/')
 }
+
+// Placeholder for form change handler
+const handleFormChange = () => {
+    console.log('Form change handler is not yet implemented.');
+};
+
+const handleFormSave = () => {
+    console.log('Form save handler is not yet implemented.');
+};
```
why this?

gianfrancodemarco commented:
@AnshSinghSonkhia thanks a lot for this! I've added a few comments; also, can you add a few tests? Thanks

AnshSinghSonkhia commented:
Oh, so sorry. I was testing in the console and forgot to remove that code.

AnshSinghSonkhia commented:
<img width="778" height="126" alt="image" src="https://github.com/user-attachments/assets/8e534278-0eab-4b0d-8aca-da178a625c9b" />

----

<img width="1015" height="247" alt="image" src="https://github.com/user-attachments/assets/d804f917-f7af-43da-911c-363baa34212f" />

AnshSinghSonkhia commented:
## Test Summary

I've successfully added comprehensive tests for the TXT export functionality in the resume builder application. Here's what was implemented:

### 1. ExporterService Tests `ExporterService.test.js`

Added the following test suites and cases:

`exportToTXT` Tests

✅ Complete resume export: Tests exporting a full resume with personal info, experiences, education, and custom sections
✅ Minimal resume export: Tests exporting a resume with just basic personal information
✅ Empty resume handling: Tests graceful handling of completely empty resume data
✅ Filename generation: Tests correct filename generation using the existing `getFilename` method
✅ Error handling: Tests graceful fallback when filename generation fails
`htmlToText` Helper Tests
✅ HTML conversion: Tests converting HTML markup to plain text
✅ Empty/null handling: Tests handling of empty or null HTML content
✅ Newline cleanup: Tests removal of excessive newlines in converted text

### 2. EditorPage Tests `EditorPage.test.js`

Added the TXT Export test suite:

✅ Successful export: Tests that `handleDownloadTXT` calls `ExporterService.exportToTXT` with correct resume data
✅ Error handling: Tests graceful error handling when TXT export fails
✅ Analytics tracking: Tests that `sa_event` is called with 'download_txt'
✅ Data passing: Tests that the correct resume data is passed to the export service

### 3. Key Features Tested

- Complete export workflow: From button click to file download
- Error scenarios: Network failures, missing data, etc.
- Data integrity: Ensuring all resume sections are included in TXT output
- File handling: Blob creation, URL generation, download triggers
- Analytics: Event tracking for user behavior analysis
- Edge cases: Empty data, missing personal info, HTML content conversion

### 4. Test Results

- 16 test files passed
- 340 total tests passed
- All new TXT export tests passing
- No regressions in existing functionality
- The tests follow the same patterns established for PDF and HTML export testing, ensuring consistency across the codebase. They provide comprehensive coverage of both happy path scenarios and error conditions, helping maintain reliable TXT export functionality.

AnshSinghSonkhia commented:
@gianfrancodemarco glad to work under your mentorship.
